Tests/GitHubCodespaces.tests.ps1

```diff
@@ -0,0 +1,450 @@
+
+    Context -Name 'When creating a repository for the authenticated user' {
+
+        Context -Name 'When creating a codespace with default settings with RepositoryId' {
+            BeforeAll {
+                $newGitHubCodespaceParms = @{
+                    RepositoryId = $repo.Id
+                }
+                $codespace = New-GitHubCodespace @newGitHubCodespaceParms
+                Start-Sleep -Seconds 2
+            }
+
+            It 'Should return an object of the correct type' {
+                $codespace | Should -BeOfType PSCustomObject
+            }
+
+            It 'Should return the correct properties' {
+                $codespace.display_name | Should -Not -BeNullOrEmpty
+                $codespace.repository.name | Should -Be $repo.name
+                $codespace.idle_timeout_minutes | Should -Be 30
+                $codespace.owner.UserName | Should -Be $script:OwnerName
+                $codespace.template | Should -BeNullOrEmpty
+            }
+
+            AfterAll {
+                if ($codespace)
+                {
+                    Remove-GitHubCodespace -CodespaceName $codespace.name -Confirm:$false
+                }
+            }
+        }
+
+        Context -Name 'When creating a codespace with default settings with Ref' {
+            BeforeAll {
+                $repoWithPR = Get-GitHubRepository -OrganizationName 'super-fake-not-real-org' |
+                    Where-Object { $_ | Get-GitHubPullRequest } |
+                    Select-Object -First 1
+                $pullRequest = $repoWithPR | Get-GitHubPullRequest | Select-Object -First 1
+                $newGitHubCodespaceParms = @{
+                    Ref = $pullRequest.head.ref
+                    RepositoryId = $repoWithPR.Id
+                }
+                $codespace = New-GitHubCodespace @newGitHubCodespaceParms
+                Start-Sleep -Seconds 2
+            }
+
+            It 'Should return an object of the correct type' {
+                $codespace | Should -BeOfType PSCustomObject
+            }
+
+            It 'Should return the correct properties' {
+                $codespace.display_name | Should -Not -BeNullOrEmpty
+                $codespace.git_status.ref | Should -Be $pullRequest.head.ref
+                $codespace.repository.name | Should -Be $repoWithPR.name
+                $codespace.idle_timeout_minutes | Should -Be 30
+                $codespace.owner.UserName | Should -Be $script:OwnerName
+                $codespace.template | Should -BeNullOrEmpty
+            }
+
+            AfterAll {
+                if ($codespace)
+                {
+                    Remove-GitHubCodespace -CodespaceName $codespace.name -Confirm:$false
+                }
+            }
+        }
+
+        Context -Name 'When creating a codespace with default settings from a PullRequest' {
+            BeforeAll {
+                  $repoWithPR = Get-GitHubRepository -OrganizationName 'super-fake-not-real-org' |
```

Comment:
Same question here.  Why would this test work at all given that this org doesn't exist?